Taskfile.yml

```diff
@@ -750,44 +769,54 @@ tasks:
   controller:kind-create-wi:
     desc: Creates a kind cluster and local Docker registry with OIDC + workload identity enabled. Images in the local registry can be pulled in the kind cluster.
     run: always
-    dir: "v2/"
     deps: 
       - az-login
     cmds:
-      - "rm -rf out/workload-identity"
-      - "mkdir -p out/workload-identity"
-      - "{{.SCRIPTS_ROOT}}/create-kind-wi-storage.sh -d out/workload-identity -p {{.TEST_LIVE_RESOURCE_PREFIX}}"
+      - "rm -rf {{.KIND_WORKLOAD_IDENTITY_PATH}}"
+      - "mkdir -p {{.KIND_WORKLOAD_IDENTITY_PATH}}"
+      - "{{.SCRIPTS_ROOT}}/create-kind-wi-storage.sh -d {{.KIND_WORKLOAD_IDENTITY_PATH}} -p {{.TEST_LIVE_RESOURCE_PREFIX}}"
       - "export KIND_CLUSTER_NAME=asov2-wi && \
-            export SERVICE_ACCOUNT_ISSUER=$(cat out/workload-identity/azure/saissuer.txt) && \
+            export SERVICE_ACCOUNT_ISSUER=$(cat {{.KIND_WORKLOAD_IDENTITY_PATH}}/azure/saissuer.txt) && \
             {{.SCRIPTS_ROOT}}/kind-with-registry.sh"
     status:
       - "kind get clusters | grep \"^asov2-wi$\""
     env:
       SERVICE_ACCOUNT_KEY_FILE:
-        sh: "echo ${HOSTROOT:-$PWD}/v2/out/workload-identity/sa.pub" # Have to use HOSTROOT here for mounting files when in docker-in-docker as paths are relative to host
+        sh: "echo ${HOSTROOT:-$PWD}/{{.KIND_WORKLOAD_IDENTITY_PATH}}/sa.pub" # Have to use HOSTROOT here for mounting files when in docker-in-docker as paths are relative to host
       SERVICE_ACCOUNT_SIGNING_KEY_FILE:
-        sh: "echo ${HOSTROOT:-$PWD}/v2/out/workload-identity/sa.key"
+        sh: "echo ${HOSTROOT:-$PWD}/{{.KIND_WORKLOAD_IDENTITY_PATH}}/sa.key"
 
   controller:create-mi-for-workload-identity:
-    desc: Creates a managed identity and federated identity credential for user in a kind cluster
-    dir: "v2/"
+    desc: Creates a managed identity and federated identity credential and stores their details in the specified path
     deps: 
       - az-login
     cmds:
-      - "{{.SCRIPTS_ROOT}}/make-mi-fic.sh -g {{.RESOURCE_GROUP}} -i {{.ISSUER}} -d ./out/workload-identity"
+      - "{{.SCRIPTS_ROOT}}/make-mi-fic.py -g {{.RESOURCE_GROUP}} -i {{.ISSUER}} -s {{.SUBJECT}} -d ./{{.DIR}}"
     vars:
+      DIR: "{{default .KIND_WORKLOAD_IDENTITY_PATH .DIR}}"
       RESOURCE_GROUP:
-        sh: "cat ./out/workload-identity/azure/rg.txt"
+        sh: |
+          if [ ! -z "{{.RESOURCE_GROUP}}" ]; then
+            echo "{{.RESOURCE_GROUP}}"
+          else
+            cat ./{{.DIR}}/azure/rg.txt
+          fi
```

Comment:
This can be done more easily using Taskfile templating. Looks something like this:

``` yaml
RESOURCE_GROUP_FILE: 
    sh: cat ./{{.DIR}}/azure/rg.txt
RESOURCE_GROUP: "{{default .RESOURCE_GROUP .RESOURCE_GROUP_FILE}}"
```